src/dbnode/x/xio/reader_slice_of_slices_from_block_readers_iterator.go

```diff
@@ -104,3 +104,7 @@ func (it *readerSliceOfSlicesIterator) Size() (int, error) {
 	}
 	return size, nil
 }
+
+func (it *readerSliceOfSlicesIterator) Rewind() {
+	it.idx = 0
```

Comment:
Same comment here:

> Looking at `Reset(...)` this should be -1?
> 
> Perhaps best to create an internal method called `resetIterationState()` and call that both from `Reset(...)` and `Rewind(...)` that sets it to -1?
> 
> Also might be worth writing a test since we didn't catch that previous to this review and would be good to regression test against.